iphone/Classes/TiUITableView.m

```diff
@@ -2283,6 +2293,57 @@ - (void)tableView:(UITableView *)ourTableView moveRowAtIndexPath:(NSIndexPath *)
   [self triggerActionForIndexPath:destinationIndexPath fromPath:sourceIndexPath tableView:ourTableView wasAccessory:NO search:NO name:@"move"];
 }
 
+#if IS_SDK_IOS_13
+- (BOOL)tableView:(UITableView *)tableView shouldBeginMultipleSelectionInteractionAtIndexPath:(NSIndexPath *)indexPath
+{
+  RETURN_IF_SEARCH_TABLE_VIEW(NO);
+
+  return [TiUtils boolValue:[[self proxy] valueForUndefinedKey:@"allowsMultipleSelectionDuringEditing"] def:NO] && [TiUtils boolValue:[[self proxy] valueForUndefinedKey:@"allowsMultipleSelectionInteraction"] def:NO];
+}
+
+- (void)tableView:(UITableView *)tableView didBeginMultipleSelectionInteractionAtIndexPath:(NSIndexPath *)indexPath
+{
+  editing = YES;
+}
+
+- (void)tableViewDidEndMultipleSelectionInteraction:(UITableView *)tableView
+{
+  if ([self.proxy _hasListeners:@"rowsselected"]) {
+    NSMutableArray *selectedItems = [NSMutableArray arrayWithCapacity:tableView.indexPathsForSelectedRows.count];
+    NSMutableDictionary *startingRowObject = [NSMutableDictionary dictionaryWithCapacity:1];
+
+    for (int i = 0; i < tableView.indexPathsForSelectedRows.count; i++) {
+      NSIndexPath *index = tableView.indexPathsForSelectedRows[i];
+      NSInteger sectionIdx = [index section];
+      NSArray *sections = [(TiUITableViewProxy *)[self proxy] internalSections];
+      TiUITableViewSectionProxy *section = [self sectionForIndex:sectionIdx];
+
+      NSInteger rowIndex = [index row];
+      int dataIndex = 0;
+      int c = 0;
+      TiUITableViewRowProxy *row = [section rowAtIndex:rowIndex];
+
+      for (TiUITableViewSectionProxy *section in sections) {
```

Comment:
This for loop get's used a lot in our table view proxy. Make a function out of it so we can reuse the code.

Reply:
Updated